patches/0005-Add-CNG-crypto-backend.patch

```diff
@@ -532,6 +527,42 @@ index bc595e91024f11..7766d674f5cfaf 100644
 +	}
 +	return true
 +}
+diff --git a/src/crypto/internal/backend/fips140/cng.go b/src/crypto/internal/backend/fips140/cng.go
+new file mode 100644
+index 00000000000000..22e4197bbb3ba5
+--- /dev/null
++++ b/src/crypto/internal/backend/fips140/cng.go
+@@ -0,0 +1,30 @@
++// Copyright 2024 The Go Authors. All rights reserved.
++// Use of this source code is governed by a BSD-style
++// license that can be found in the LICENSE file.
++
++//go:build goexperiment.cngcrypto
++
++package fips140
++
++import (
++	"internal/syscall/windows/sysdll"
++	"syscall"
++	"unsafe"
++)
++
++// Don't use github.com/microsoft/go-crypto-winnative here.
++// The fips140 package should have minimal dependencies.
++// Also, don't directly query the system FIPS mode from the registry,
++// there are some no-longer documented legacy entries that can enable FIPS mode,
++// and BCryptGetFipsAlgorithmMode supports them all.
++var (
++	bcrypt = syscall.MustLoadDLL(sysdll.Add("bcrypt.dll"))
++
++	bcryptGetFipsAlgorithmMode = bcrypt.MustFindProc("BCryptGetFipsAlgorithmMode")
++)
++
++func systemFIPSMode() bool {
++	var enabled uint32
++	bcryptGetFipsAlgorithmMode.Call(uintptr(unsafe.Pointer(&enabled)))
```

Comment:
Handle an error here? (In case the call puts nonsense into `enabled` in some error case?)